### ISSUE TITLE
Container still running after HealthCheck failure

### ISSUE THREAD

**Issue by royprigat** [testcontainers/testcontainers-node#141]
Attempting to start a new container:
```
container = await new GenericContainer(DOCKER_IMAGE)
    .withWaitStrategy(Wait.forHealthCheck())
    .start();
```

If health check fails, the container is still running in the background and the object returned is 'undefined'.
This means I can't use container.stop() to actually stop the container.

Is there any way to stop it completely?

**Comment by cristianrgreco** [comment]
Hi @royprigat, thanks for raising the issue, I've created a PR for this, let me know if you want to take a look else I'll release it

**Comment by cristianrgreco** [comment]
Released in 4.2.0
